### ISSUE TITLE
default htb qdisc configured on cbr0 sometimes drops all packets

### ISSUE THREAD

**Issue by mikedanese** [kubernetes/kubernetes#25092]
I'm not sure where the defaults come from but they sometimes drop all packets causing container IPs to become unreachable. ARP fails.
```console
root@k-0-minion-c9mq:/home/mikedanese# tc qdisc replace dev cbr0 root htb
root@k-0-minion-c9mq:/home/mikedanese# ping 10.244.0.2                                                                                                                                                                                 
PING 10.244.0.2 (10.244.0.2) 56(84) bytes of data.
From 10.244.0.1 icmp_seq=1 Destination Host Unreachable
From 10.244.0.1 icmp_seq=2 Destination Host Unreachable
From 10.244.0.1 icmp_seq=3 Destination Host Unreachable
root@k-0-minion-c9mq:/home/mikedanese# tc -p -s qdisc show dev cbr0
qdisc htb 8001: root refcnt 2 r2q 10 default 0 direct_packets_stat 0 direct_qlen 0
 Sent 0 bytes 0 pkt (dropped 9, overlimits 0 requeues 0) 
 backlog 0b 0p requeues 0
root@k-0-minion-c9mq:/home/mikedanese# tc qdisc replace dev cbr0 root pfifo_fast
root@k-0-minion-c9mq:/home/mikedanese# ping 10.244.0.2                                                                                                                                                                                 
PING 10.244.0.2 (10.244.0.2) 56(84) bytes of data.
64 bytes from 10.244.0.2: icmp_seq=1 ttl=64 time=0.106 ms
64 bytes from 10.244.0.2: icmp_seq=2 ttl=64 time=0.066 ms
```
I'm not sure if this is related but I also noticed on this machine that the default pfifo qdisc drops all packets. pfifo is the default qdisc for an htb node so it's used in the unconfigured htb qdisc:
```console
root@k-0-minion-c9mq:/home/mikedanese# tc qdisc replace dev cbr0 root pfifo
root@k-0-minion-c9mq:/home/mikedanese# ping 10.244.0.2
PING 10.244.0.2 (10.244.0.2) 56(84) bytes of data.
From 10.244.0.1 icmp_seq=1 Destination Host Unreachable
From 10.244.0.1 icmp_seq=2 Destination Host Unreachable
From 10.244.0.1 icmp_seq=3 Destination Host Unreachable
root@k-0-minion-c9mq:/home/mikedanese# ping 10.244.0.2^C
root@k-0-minion-c9mq:/home/mikedanese# tc -p -s qdisc show dev cbr0
qdisc pfifo 8004: root refcnt 2 limit 0p
 Sent 0 bytes 0 pkt (dropped 18, overlimits 0 requeues 0) 
 backlog 0b 0p requeues 0 
```
This is because the default value for queue size is zero. On other machines I've seen the default value set to 1 packet and 1000 packets. When I set the limit explicitly the container becomes reachable again.
```console
root@k-0-minion-c9mq:/home/mikedanese# tc qdisc replace dev cbr0 root pfifo limit 1000
root@k-0-minion-c9mq:/home/mikedanese# ping 10.244.0.2                                                                                                                                               
PING 10.244.0.2 (10.244.0.2) 56(84) bytes of data.
64 bytes from 10.244.0.2: icmp_seq=1 ttl=64 time=0.078 ms
64 bytes from 10.244.0.2: icmp_seq=2 ttl=64 time=0.054 ms
```

The broken configuration is applied by kubelet to implement traffic shapping and there is no option to turn this off. This patch fixes the kubelet on broken machines by disabling traffic shapping:
https://gist.github.com/mikedanese/31d1e60e2659180fad4d352cce30dead

cc @dcbw @freehan @kubernetes/sig-network @kubernetes/goog-cluster

**Comment by dcbw** [comment]
@mikedanese does this happen with --configure-cbr0 and packet shaping too?  The kubenet implementation just uses the existing kubernetes shaping code that --configure-cbr0 does.

**Comment by dcbw** [comment]
It seems the *fifo qdiscs default their packet limit to the txquelen of the interface (man tc-pfifo).  It also seems most virtual interface types (bridge, etc) set a dev->tx_queue_len of 0.  It further seems that pkg/util/bandwidth/linux.go relies on the default packet limit.  I could swear this worked before with kubenet, and it must have worked with --configure-cbr0 when the code was originally added.

I'll check around to figure out what the default limit should be for interfaces that don't specify a tx queue length.

**Comment by dcbw** [comment]
Note that to test with --configure-cbr0 you'd also need --reconcile-cidr.

**Comment by dcbw** [comment]
Upstream CNI fix is here: https://github.com/containernetworking/cni/pull/199 and we'll also work around it in kubenet for now.

**Comment by dcbw** [comment]
FWIW, the upstream CNI fix got merged for CNI 0.3.0
